apps/remote_control/test/lexical/remote_control/code_mod/rename_test.exs

```diff
@@ -509,6 +553,180 @@ defmodule Lexical.RemoteControl.CodeMod.RenameTest do
+    end
+
+    test "succeeds when the cursor is at the definition with 1 arity" do
+      {:ok, result} =
+        ~q[
+        defmodule Foo do
+          def |bar(x) do
+          end
+        end
+      ] |> rename("renamed")
+
+      assert result =~ ~S[def renamed(x)]
+    end
+
+    test "succeeds when the function has default params" do
+      {:ok, result} =
+        ~q[
+        defmodule Foo do
+          def |bar(x \\ 1) do
+          end
+        end
+      ] |> rename("renamed")
+
+      assert result =~ ~S[def renamed(x \\ 1)]
+    end
+
+    test "succeeds when the function has declaring defaults" do
+      {:ok, result} =
+        ~q[
+        defmodule Foo do
+          def |bar(x, y \\ 1)
+
+          def bar(x, y) do
+          end
+        end
+      ] |> rename("renamed")
+
+      assert result =~ ~S[def renamed(x, y \\ 1)]
+      assert result =~ ~S[def renamed(x, y) do]
+    end
+
+    test "succeeds when the cursor is at the reference for a local call" do
+      {:ok, result} =
+        ~q/
+        defmodule Foo do
+          def bar do
+          end
+
+          def baz do
+            [|bar()]
+          end
+        end
+      / |> rename("renamed")
+
+      assert result =~ ~S/[renamed()]/
+    end
+
+    test "it should rename both definition and reference" do
+      {:ok, result} =
+        ~q/
+        defmodule DefinitionModule do
+          def |hello do
+            :world
+          end
+        end
+
+        defmodule DefinitionModuleTest do
+          use ExUnit.Case
+
+          test "it should call the |hello function" do
+            assert DefinitionModule.hello() == :world
+          end
+        end
+      / |> rename("renamed")
+
+      assert result =~ ~S/def renamed do/
+      assert result =~ ~S/assert DefinitionModule.renamed() == :world/
+    end
+
+    @tag :skip
+    @doc """
+    There is a bug in the `Entity.resolve/2` function that is causing the test to fail.
+    """
+    test "succeeds when the cursor is at the reference for a local call with capture" do
+      {:ok, result} =
+        ~q{
+        defmodule Foo do
+          def bar(x) do
+            x
+          end
+
+          def baz do
+            [&|bar/1]
+          end
+        end
+      } |> rename("renamed")
+
+      assert result =~ ~S/[renamed()]/
+    end
+
+    test "succeeds when the cursor is at the reference for a qualified call with multiple arguments" do
+      {:ok, result} =
+        ~q[
+        defmodule DefinitionModule do
+          def bar(x, y) do
+            {x, y}
+          end
+        end
+
+        defmodule Bar do
+          DefinitionModule.|bar(1, 2)
+        end
+      ] |> rename("renamed")
+
+      assert result =~ ~S[def renamed(x, y) do]
+      assert result =~ ~S[DefinitionModule.renamed(1, 2)]
+    end
+
+    @tag :skip
+    @doc """
+    This is a bug in our function indexation:
+      when the cursor is at the reference for a imported call, the definition function can't be found.
```

Comment:
why not fix the bug?

Reply:
Still wip, so I use test record the issue first